lib/block_view_test.go

```diff
@@ -14690,3 +15303,1854 @@ func TestUpdateProfileChangeBack(t *testing.T) {
+		bidAmountMinusRoyalties := bidAmountNanos - expectedCoinRoyalty - expectedCreatorRoyalty
+
+		m0BalAfter := _getBalance(t, chain, nil, m0Pub)
+		require.Equal(m0BalBefore+expectedCreatorRoyalty, m0BalAfter)
+		require.Equal(uint64(1122), m0BalAfter)
+
+		m1BalAfter := _getBalance(t, chain, nil, m1Pub)
+		require.Equal(m1BalBefore-2+bidAmountMinusRoyalties, m1BalAfter)
+		require.Equal(uint64(11865), m1BalAfter)
+
+		// Creator coin.
+		bitCloutLocked, _ := _getCreatorCoinInfo(t, db, params, m0Pub)
+		require.Equal(m0InitialBitCloutLocked+uint64(1011), bitCloutLocked)
+	}
+
+	// Error case: Let's make sure that no royalties are paid if there are no coins in circulation.
+	{
+		_, coinsInCirculationNanos := _getCreatorCoinInfo(t, db, params, m0Pub)
+		require.Equal(uint64(30365901), coinsInCirculationNanos)
+
+		// Sell all the coins.
+		_creatorCoinTxnWithTestMeta(
+			testMeta,
+			10,     /*feeRateNanosPerKB*/
+			m0Pub,  /*updaterPkBase58Check*/
+			m0Priv, /*updaterPrivBase58Check*/
+			m0Pub,  /*profilePubKeyBase58Check*/
+			CreatorCoinOperationTypeSell,
+			0,                       /*BitCloutToSellNanos*/
+			coinsInCirculationNanos, /*CreatorCoinToSellNanos*/
+			0,                       /*BitCloutToAddNanos*/
+			0,                       /*MinBitCloutExpectedNanos*/
+			0,                       /*MinCreatorCoinExpectedNanos*/
+		)
+
+		// Create a bid on <post1, #9>, which is still for sale.
+		bidAmountNanos := uint64(100)
+		serialNumber := uint64(9)
+		bidEntries := DBGetNFTBidEntries(db, post1Hash, serialNumber)
+		require.Equal(0, len(bidEntries))
+
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m3Pub,
+			m3Priv,
+			post1Hash,
+			serialNumber,   /*SerialNumber*/
+			bidAmountNanos, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post1Hash, serialNumber)
+		require.Equal(1, len(bidEntries))
+
+		// Balance before.
+		m0BalBefore := _getBalance(t, chain, nil, m0Pub)
+		require.Equal(uint64(2159), m0BalBefore)
+
+		_acceptNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m0Pub,
+			m0Priv,
+			post1Hash,
+			serialNumber, /*SerialNumber*/
+			m3Pub,        /*bidderPkBase58Check*/
+			bidAmountNanos,
+			"", /*UnencryptedUnlockableText*/
+		)
+
+		// Check royalties. 10% for the creator, 10% for the coin.
+		expectedCreatorRoyalty := bidAmountNanos / 10
+		require.Equal(uint64(10), expectedCreatorRoyalty)
+		expectedCoinRoyalty := bidAmountNanos / 10
+		require.Equal(uint64(10), expectedCoinRoyalty)
+		bidAmountMinusRoyalties := bidAmountNanos - expectedCoinRoyalty - expectedCreatorRoyalty
+
+		m0BalAfter := _getBalance(t, chain, nil, m0Pub)
+		require.Equal(m0BalBefore-2+bidAmountMinusRoyalties+expectedCreatorRoyalty, m0BalAfter)
+		require.Equal(uint64(2247), m0BalAfter)
+
+		// Creator coin --> Make sure no royalties were added.
+		bitCloutLocked, _ := _getCreatorCoinInfo(t, db, params, m0Pub)
+		require.Equal(uint64(0), bitCloutLocked)
+	}
```

Comment:
should add test to make sure min bid amount nanos is behaving correctly.

Reply:
See "TestNFTMoreErrorCases." I made an artistic decision to allow NFT owners to accept standing offers (serial #0 bids) even if the offer is below the min bid amount.  If this were not the case, owners would have to broadcast 3 transactions to achieve the same effect: (1) update the NFT to be not for sale, (2) update the NFT to be for sale with a lower bid amount, (3) accept the min bid amount.  Seems better to just let them accept the standing offers regardless of min bid amount.  Let me know if you disagree.